src/count_loc.jl

```diff
@@ -69,5 +83,184 @@ end
 count_readme(table) = count_docs(table, ("readme", "readme.md"))
 
 count_julia_loc(pkg::Package, args...) = count_julia_loc(pkg.lines_of_code, args...)
+count_docstrings(pkg::Package, args...) = count_docstrings(pkg.lines_of_code, args...)
 count_docs(pkg::Package, args...) = count_docs(pkg.lines_of_code, args...)
 count_readme(pkg::Package, args...) = count_readme(pkg.lines_of_code, args...)
+
+#####
+##### Custom Julia line counting
+#####
+
+# We can't do this at the level of SyntaxNode's because we've lost whitespace & comments already
+# So we will go to GreenNode's
+
+# Avoid piracy by defining AbstractTrees definitions on a wrapper
+struct GreenNodeWrapper
+    node::JuliaSyntax.GreenNode
+    source::JuliaSyntax.SourceFile
+end
+
+function AbstractTrees.children(wrapper::GreenNodeWrapper)
+    return map(n -> GreenNodeWrapper(n, wrapper.source), JuliaSyntax.children(wrapper.node))
+end
+
+function parse_green_one(file_path)
+    file = read(file_path, String)
+    @debug(string("Parsing ", file_path))
+    parsed = JuliaSyntax.parse(JuliaSyntax.GreenNode, file; ignore_trivia=false)
+    return GreenNodeWrapper(parsed[1], JuliaSyntax.SourceFile(file; filename=basename(file_path)))
+end
+
+# Module to make it easier w/r/t/ import clashes
+module CategorizeLines
+
+using JuliaSyntax: GreenNode, is_trivia, haschildren, is_error, children, span, SourceFile, source_location
+
+struct NodeSummary
+    starting_line::Int
+    ending_line::Int
+    summary::String
+end
+
+# Based on the recursive printing code for GreenNode's
+# Here, instead of printing, we collect all of the objects
+# that start on a given line.
+function categorize_lines!(d, node, source, nesting=0, pos=1)
+    starting_line, _ = source_location(source, pos)
+    v = get!(Vector{NodeSummary}, d, starting_line)
+
+    is_leaf = !haschildren(node)
+    if !is_leaf
+        new_nesting = nesting + 1
+        p = pos
+        for x in children(node)
+            categorize_lines!(d, x, source, new_nesting, p)
+            p += x.span
+        end
+        ending_line, _ = source_location(source, p)
+    else
+        ending_line = starting_line
+    end
+    push!(v, NodeSummary(starting_line, ending_line, summary(node),
+    # is_error=is_error(node), is_leaf, is_trivia=is_trivia(node), nesting
+    ))
+    return nothing
+end
+
+export NodeSummary
+end
+
+using .CategorizeLines
+
+# TODO:
+# Handle `@doc` calls?
```

Comment:
I don't think `@doc` matters. It's rarely used and supporting it won't matter much for aggregate package stats. And when it is used, it can be used for things *other* than attaching docstrings to objects. The `@doc` api is bizarre if you ask me :)